### ISSUE TITLE
editor accessibility must be reenabled at the start of each lesson

### ISSUE THREAD

**Issue by devinprater** [freeCodeCamp/freeCodeCamp#18487]
**Looking forward for reporting a security issue:**
Please report security issues by sending an email to `security@freecodecamp.org` instead of raising a Github issue.

**Describe the bug**
The Monaco editor contains features for accessibility, specifically a mode for screen reader users which helps in navigating and using the editor effectively. This feature, as of now, must be enabled at the start of each lesson, even after enabling it once at the beginning of the session.
**To Reproduce**
1. Go to any lesson in the curriculum.
2. Enter the code editor, set focus to it.
3. Press alt+f1 to show accessibility help for the editor, and choose to enable the screen reader mode
4. complete the lesson, advance to the next, and repeat steps two and three

**Expected behavior**
The screen reader mode should have stayed enabled across lessons.
**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] 69.0.3497.100 (Official Build) (64-bit)

**Smartphone (please complete the following information):**

**Additional context**
Tested screen reader: NVDA version 2018.3.2
from https://www.nvaccess.org/download/

**Comment by iAziz786** [comment]
Hey, @devinprater @Bouncey. As a part of hacktoberfest, I would like to work on this issue.

Can I expect mentorship on this?

**Comment by kelvinromero** [comment]
@iAziz786 I'm new on this too. But if you need help, I would like to work on this issue too.

**Comment by iAziz786** [comment]
@kelvinromero Can you reproduce this issue?

**Comment by saransha** [comment]
A group of us are working on this issue for a class project in university. We aim to address this in the next couple of days. We will try and at least present the community with a first iteration of our proposed solution. 

Just a heads up. If the current developers have any advice or tips, feel free to contact me. Thank you.

**Comment by moT01** [comment]
Any updates on this @oalhait @saransha ?

**Comment by moT01** [comment]
This looks to be fixed, the editor is focused on load. Can you confirm that this is fixed @devinprater ? If so, this can probably be closed.

**Comment by ojeytonwilliams** [comment]
@moT01 I don't think it is fixed.  It focuses on load by default, but I think that turning on the screen reader mode is separate from just focusing.

I don't have a screen reader installed, but if I follow the steps above I keep having to re-enable that mode each time I go to a new challenge.

**Comment by devinprater** [comment]
Sorry, I don’t think the editor should be focused on load, as that means the screen reader user most unfocus it, move back to the heading to read the challenge, then focus it again to write code. Also, I still have to enable accessibility mode on each challenge.

**Comment by devinprater** [comment]
VoiceOver on MacOS is… not the best, sadly, when it comes to web apps. If possible, you can send technical info about this to accessibility@apple.com <mailto:accessibility@apple.com>

Mono is Microsoft’s editor, which seems to be VSCode stripped down to the editor. Perhaps you could contact Microsoft and see if they could add a way to make accessibility states persistent across loads, if this isn’t possible already.

**Comment by ojeytonwilliams** [comment]
> Sorry, I don’t think the editor should be focused on load, as that means the screen reader user most unfocus it, move back to the heading to read the challenge, then focus it again to write code.

That is annoying.  For now you should be able to focus the description by pressing escape (at the moment, it's still in a state of flux), but I'll see about making screen-reader mode something that persists between challenges.

**Comment by moT01** [comment]
Seeing as how this issue is beyond our control, should we close this and make a new issue about the editor being focused on load?

You can feel free to email apple or microsoft if you want @devinprater. Thanks for the advice though.

**Comment by ojeytonwilliams** [comment]
@moT01 no need to close this. I think I can fix it.

**Comment by devinprater** [comment]
Yes, we can close this issue.

**Comment by ojeytonwilliams** [comment]
Quick update.  Both the auto-focusing and the persistence of accessibility mode *should* be fixable.  I think I'm about 90% of the way there, but for some reason it stops letting you toggle the accessibility support after a couple of toggles.  

This, obviously, isn't ideal, but most of the functionality is there.
